indico/modules/events/contributions/controllers/management.py

```diff
@@ -596,3 +612,45 @@ def _process(self):
             abstracts_settings.set(self.event_new, 'description_settings', form.data)
             return jsonify_data(flash=False)
         return jsonify_form(form)
+
+class RHCreateReferenceMixin:
+    """Common methods for RH class creating a ContibutionReference or SubContributionReference"""
+
+    def _checkParams(self):
+        self.reference_value = request.form['value']
+        reference_type_name = request.form['type']
+        self.reference_type = ReferenceType.find_one(name=reference_type_name)
+
+    @staticmethod
+    def jsonify_reference(reference):
+        return jsonify(id=reference.id)
+
+
+class RHCreateContributionReferenceREST(RHCreateReferenceMixin, RHManageContributionBase):
+    """REST endpoint to add a reference to a Contribution"""
+
+    def _checkParams(self, params):
+        RHManageContributionBase._checkParams(self, params)
+        RHCreateReferenceMixin._checkParams(self)
+
+    def _process_POST(self):
+        reference = ContributionReference(reference_type=self.reference_type, value=self.reference_value,
+                                          contribution_id=self.contrib.id)
```

Comment:
no need to set that id, appending it to the relationship does that for you on flush